apps/src/studio/studio.js

```diff
@@ -1428,6 +1428,15 @@ var arrangeStartBlocks = function (config) {
   }
 };
 
+function extendHandleClearPuzzle() {
+  var orig = studioApp.handleClearPuzzle.bind(studioApp);
+  studioApp.handleClearPuzzle = function (config) {
+    orig(config);
+    studioApp.resetButtonClick();
+    annotationList.clearRuntimeAnnotations();
+  };
+}
+
```

Comment:
With multiple apps now doing this (studio/applab), I wonder if we can come up with a cleaner, more general solution. Maybe as part of the config we pass into init we give the ability to provide an optional onClearPuzzle?